README.md
<h1 align="center">Welcome to cookies package 👋</h1>

<p align="center">
  <img src="https://sonarcloud.io/api/project_badges/measure?project=Callumgm_Cookies_Package&metric=ncloc">
  <img src="https://img.shields.io/badge/version-1.2.3-blue.svg?cacheSeconds=2592000" >
  <img src="https://img.shields.io/badge/Maintained%3F-yes-green.svg" >
  <img src="https://img.shields.io/badge/license-MIT-yellow.svg" >
  <img src="https://img.shields.io/github/last-commit/Callumgm/Cookies_Package">
  <a href="https://twitter.com/Flashouttt" target="_blank">
    <img src="https://img.shields.io/twitter/follow/Flashouttt.svg?style=social">
  </a>
</p>

> Python package for easier use of certin functions

## 🚀 〢 Usage

```sh-session
pip install cookies_package
```

## 📃 〢 Documentation
```py
clear() = Clears the console / Syntax = clear()

setTitle() = Set console title / Syntax = setTitle("Hello world", "CookiesKush420")

slowPrint() = Print text letter by letter / Syntax = slowPrint("Hello world", 0.04)

curl_download() = Download file from github private repo using curl / Syntax = curl_download("main.py", privateToken, "raw.githubusercontent.com/Callumgm/test/master/main.py")

obfusacate() = obfuscate file code before compiling from temp folder / Syntax = obfusacate("main")
```

## 💭 〢 ChangeLog
```diff
v1.2.3 ⋮ 2022-07-01
+ fixed loading animation printing breaking

v1.2.2 ⋮ 2022-07-01
+ fixed shitty import errors
+ created sections
- removed seperate files and place all code into main.py

v1.2.1 ⋮ 2022-07-01
+ small hotfix

v1.1.3 ⋮ 2022-07-01
+ added loading antimation to obfusacate and curl commands
+ cleaned code
- removed obfuscate code from main into utils

v1.0.3 ⋮ 2022-07-01
+ added obfuscate code with random key
+ added download github code via curl
+ added loading animation

v0.0.1 ⋮ 2022-05-23
+ initial release
```

## 👤 〢 Author

 👤 **CookiesKush420**  
- Website: http://cookiesservices.xyz/  
- Twitter: [@Flashouttt](https://twitter.com/Flashouttt)  
- GitHub: [@Callumgm](https://github.com/Callumgm)    


## 🤝 〢 Contributing
Contributions, issues and feature requests are welcome!<br />Feel free to check
[issues page](https://github.com/Callumgm/Cookies_Package/issues).  


## 🌟 〢 Show your support
Give a ⭐️ if this project helped you! 


## 📝 〢 License
 Copyright © 2022
[CookiesKush420](https://github.com/Callumgm).<br />  This project is [MIT](https://github.com/Callumgm/Cookies_Package/blob/master/LICENCE) licensed. 


CHANGELOG.txt
Change Log
==========

0.0.1 Beta (23/05/2022)
-------------------
- First Release

1.0.0 (01/07/2022)
-------------------
- First Stable Release
- Added obfuscate code with random key
- Added download github code via curl
- Added loading animation

cookies_package/util/obfuscate.py

import os, shutil, base64, random
from Crypto.Cipher import AES
from util.loading_animation import *
from Crypto import Random

temp = os.getenv('TEMP')

def obfuscate_code(fileName):
    IV = Random.new().read(AES.block_size)                       # Generate random IV
    key = u''
    for i in range(8):
        key = key + chr(random.randint(0x4E00, 0x9FA5))          # Generate random key

    with open(f'{temp}\\{fileName}.py') as f:                    # Open file and fix imports
        _file = f.read()
        imports = ''
        input_file = _file.splitlines()
        for i in input_file:
            if i.startswith("import") or i.startswith("from"):
                imports += i+';'

    with open(f'{temp}\\{fileName}.py', "wb") as f:              # Write file with fixed imports & AES encrypted
        encodedBytes = base64.b64encode(_file.encode())
        obfuscatedBytes = AES.new(key.encode(), AES.MODE_CFB, IV).encrypt(encodedBytes)
        f.write(f'from Crypto.Cipher import AES;{imports}exec(__import__(\'\\x62\\x61\\x73\\x65\\x36\\x34\').b64decode(AES.new({key.encode()}, AES.MODE_CFB, {IV}).decrypt({obfuscatedBytes})).decode())'.encode())
    
    shutil.move(f"{temp}\\{fileName}.py", f"{os.getcwd()}\\{fileName}.py")  # Move file to main folder ready for compiling

cookies_package/main.py
import os, ctypes, sys, itertools, threading, time, shutil, base64, random
from Crypto.Cipher import AES
from Crypto import Random
from time import sleep
 
temp = os.getenv('TEMP')


'''
MODULES SECTION
'''

def obfuscate_code(fileName):
    IV = Random.new().read(AES.block_size)                       # Generate random IV
    key = u''
    for i in range(8):
        key = key + chr(random.randint(0x4E00, 0x9FA5))          # Generate random key

    with open(f'{temp}\\{fileName}.py') as f:                    # Open file and fix imports
        _file = f.read()
        imports = ''
        input_file = _file.splitlines()
        for i in input_file:
            if i.startswith("import") or i.startswith("from"):
                imports += i+';'

    with open(f'{temp}\\{fileName}.py', "wb") as f:              # Write file with fixed imports & AES encrypted
        encodedBytes = base64.b64encode(_file.encode())
        obfuscatedBytes = AES.new(key.encode(), AES.MODE_CFB, IV).encrypt(encodedBytes)
        f.write(f'from Crypto.Cipher import AES;{imports}exec(__import__(\'\\x62\\x61\\x73\\x65\\x36\\x34\').b64decode(AES.new({key.encode()}, AES.MODE_CFB, {IV}).decrypt({obfuscatedBytes})).decode())'.encode())
    
    shutil.move(f"{temp}\\{fileName}.py", f"{os.getcwd()}\\{fileName}.py")  # Move file to main folder ready for compiling


def animate_loading(method):
    def animated():
        done = False
        def animate():
            time.sleep(0.000001)
            for c in itertools.cycle(['|', '/', '-', '\\']):
                if done:
                    break
                sys.stdout.write('\r Loading ' + c)
                sys.stdout.flush()
                time.sleep(0.1)
            sys.stdout.write('\r Done!       ')
            os.system('cls')
            time.sleep(0.5)
            
        t = threading.Thread(target=animate)
        t.start()
 
        method()
        done = True
    return animated



'''
COMMANDS SECTION
'''

def clear():
    '''
    Clear console
    '''
    os.system('cls')


def setTitle(str, creator):
    '''
    Set console title with creator name
    
    Syntax = "Hello World, CookiesKush420"
    '''
    ctypes.windll.kernel32.SetConsoleTitleW(f"{str} | {creator}")


def slowPrint(str, speed):
    '''
    Print text letter by letter
    
    Syntax = "Hello World, 0.04"
    '''
    for letter in str:
        sys.stdout.write(letter);sys.stdout.flush();sleep(speed)


def curl_download(file_location_with_extention, github_token, url):
    '''
    Download file from github using curl
    
    Syntax = "main.py, github_token, raw.githubusercontent.com/Callumgm/test/master/main.py"
    '''
    animate_loading(lambda: os.system(f'curl -s -o {file_location_with_extention} https://{github_token}@{url}'))()


def obfusacate(fileName):
    '''
    obfuscate file code before compiling from temp folder
    
    Syntax = "main"
    '''
    animate_loading(lambda: obfuscate_code(fileName))()

